src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -60,210 +69,280 @@ private void captureOs() {
-            if (gitHubRepository.isPresent() && gitHubRunId.isPresent()) {
-                buildScan.link("GitHub Actions build", "https://github.com/" + gitHubRepository.get() + "/actions/runs/" + gitHubRunId.get());
+        @Override
+        public void execute(BuildResult buildResult) {
+            if (CiUtils.isJenkins(providers) || CiUtils.isHudson(providers)) {
+                Optional<String> buildUrl = Utils.envVariable("BUILD_URL", providers);
+                Optional<String> buildNumber = Utils.envVariable("BUILD_NUMBER", providers);
+                Optional<String> nodeName = Utils.envVariable("NODE_NAME", providers);
+                Optional<String> jobName = Utils.envVariable("JOB_NAME", providers);
+                Optional<String> stageName = Utils.envVariable("STAGE_NAME", providers);
+
+                buildUrl.ifPresent(url ->
+                        buildScan.link(CiUtils.isJenkins(providers) ? "Jenkins build" : "Hudson build", url));
+                buildNumber.ifPresent(value ->
+                        buildScan.value("CI build number", value));
+                nodeName.ifPresent(value ->
+                        customValueSearchLinker.addCustomValueAndSearchLink("CI node", value));
+                jobName.ifPresent(value ->
+                        customValueSearchLinker.addCustomValueAndSearchLink("CI job", value));
+                stageName.ifPresent(value ->
+                        customValueSearchLinker.addCustomValueAndSearchLink("CI stage", value));
+
+                jobName.ifPresent(j -> buildNumber.ifPresent(b -> {
+                    Map<String, String> params = new LinkedHashMap<>();
+                    params.put("CI job", j);
+                    params.put("CI build number", b);
+                    customValueSearchLinker.registerLink("CI pipeline", params);
+                }));
             }
-            Utils.envVariable("GITHUB_WORKFLOW", providers).ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI workflow", value));
-            Utils.envVariable("GITHUB_RUN_ID", providers).ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI run", value));
-        }
 
-        if (CiUtils.isGitLab(providers)) {
-            Utils.envVariable("CI_JOB_URL", providers).ifPresent(url ->
-                buildScan.link("GitLab build", url));
-            Utils.envVariable("CI_PIPELINE_URL", providers).ifPresent(url ->
-                buildScan.link("GitLab pipeline", url));
-            Utils.envVariable("CI_JOB_NAME", providers).ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI job", value));
-            Utils.envVariable("CI_JOB_STAGE", providers).ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI stage", value));
-        }
+            if(CiUtils.isTeamCity(providers)) {
```

Comment:
```suggestion
            if (CiUtils.isTeamCity(providers)) {
```
let's fix the formatting from the previous PR